### PR TITLE
 ci: fix canary-arm64 job

### PR DESCRIPTION
**Pull request by xin3liang**
Since the below commit, addons are moved out of deploy_cluster().
31d55b90fef8 ("test: mv canary test specific resources out of deploy_cluster()")

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: .github/workflows/daily-nightly-jobs.yml [hunk 1]
```diff
@@ -80,6 +80,7 @@ jobs:
           yq write -d0 -i deploy/examples/cluster-test.yaml "spec.healthCheck.livenessProbe.mgr.disabled" true
           yq write -d0 -i deploy/examples/cluster-test.yaml "spec.healthCheck.livenessProbe.osd.disabled" true
           tests/scripts/github-action-helper.sh deploy_cluster
+          tests/scripts/github-action-helper.sh deploy_all_additional_resources_on_cluster
```
Comment (travisn) [inline .github/workflows/daily-nightly-jobs.yml:83]: Good catch, the test wasn't even failing even though the additional resources were not being created.

File: .github/workflows/daily-nightly-jobs.yml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 name: Daily nightly jobs
 on:
-  schedule:
-    - cron: "0 0 * * *" # every day at midnight
-  workflow_dispatch: {}
+  pull_request:
```
Comment (travisn) [inline .github/workflows/daily-nightly-jobs.yml:3]: The tests look good, there is a known issue with the rgw multisite test that is not related to this change. So we can merge after you revert this commit.

Comment (xin3liang) [reply .github/workflows/daily-nightly-jobs.yml]: Done. I have already reverted the commit.

**Review by subhamkrai — APPROVED** [review]
